flocker/control/_model.py

```diff
@@ -418,18 +416,16 @@ def __invariant__(self):
                     return (False, '%r manifestation is not on node' % (app,))
         return (True, "")
 
-    def __new__(cls, **kwargs):
+    def __new__(cls, hostname=None, **kwargs):
         # PRecord does some crazy stuff, thus _precord_buckets; see
         # PRecord.__new__.
         if "uuid" not in kwargs and "_precord_buckets" not in kwargs:
             warn("UUID is required, this is for backwards compat with existing"
                  " tests only. If you see this in production code that's "
                  "a bug.", DeprecationWarning, stacklevel=2)
```

Comment:
In [2]: NodeState()
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-c81282e22251> in <module>()
----> 1 NodeState()

/home/richard/projects/HybridLogic/flocker/flocker/control/_model.pyc in __new__(cls, **kwargs)
    659                  " tests only. If you see this in production code that's "
    660                  "a bug.", DeprecationWarning, stacklevel=2)
--> 661             kwargs["uuid"] = ip_to_uuid(kwargs["hostname"])
    662         return PRecord.__new__(cls, **kwargs)
    663 

KeyError: 'hostname'